src/main/java/de/tum/p2p/proto/message/onion/forwarding/OnionTunnelBuildMessage.java

```diff
@@ -0,0 +1,27 @@
+package de.tum.p2p.proto.message.onion.forwarding;
+
+import de.tum.p2p.api.netty.forwarding.OnionApiMessage;
+import de.tum.p2p.proto.message.MessageType;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import java.nio.ByteBuffer;
+
+@ToString @EqualsAndHashCode(callSuper = true)
+public class OnionTunnelBuildMessage extends OnionApiMessage {
+
+    public OnionTunnelBuildMessage() {
+        super(MessageType.ONION_TUNNEL_BUILD);
+    }
+
+    public static OnionTunnelBuildMessage me() { return new OnionTunnelBuildMessage(); }
+
+    @Override
+    protected ByteBuffer writeMessage(ByteBuffer typedMessageBuffer) {
+        return typedMessageBuffer; // Onion Cover message carries no payload
```

Comment:
If you dont expect this to be used/useful throw new UnsupportedOperationException() here